themes/bootstrap3/js/searchbox_controls.js

```diff
@@ -0,0 +1,143 @@
+/*global VuFind */
+VuFind.register('searchbox_controls', function SearchboxControls() {
+  let _KeyboardClass;
+  let _KeyboardLayoutClass;
+
+  let _textInput;
+  let _resetButton;
+
+  let _enabled = false;
+  let _keyboard;
+  const _defaultTheme = "hg-theme-default";
+  const _display = {
+    "{bksp}": "&#10229;",
+    "{enter}": "&#8626;",
+    "{shift}": "&#8679;",
+    "{tab}": "&#8633;",
+    "{lock}": "&#8681;",
+  };
+
+  function _showKeyboard() {
+    if (_enabled) {
+      _keyboard.setOptions({
+        theme: `${_defaultTheme} show-keyboard`
+      });
+    }
+  }
+
+  function _hideKeyboard() {
+    _keyboard.setOptions({
+      theme: _defaultTheme
+    });
+  }
+
+  function _onChange(input){
+    _textInput.value = input;
+    if (_textInput.value === '') {
+      _resetButton.hide();
+    } else {
+      _resetButton.show();
+    }
+    if ( typeof _keyboard !== 'undefined') {
+      _keyboard.setInput(input);
+    }
+    _textInput.dispatchEvent(new Event('input'));
+  }
+
+  function _onKeyPress(button){
+    if (button === "{shift}" || button === "{lock}") {
+      let currentLayoutType = _keyboard.options.layoutName;
+      _keyboard.setOptions({
+        layoutName: currentLayoutType === "default" ? "shift" : "default"
+      });
+    }
+
+    if (button === "{enter}") {
+      document.getElementById("searchForm").submit();
+    }
+  }
+
+  function _updateKeyboardLayout(layoutName){
```

Comment:
Minor point of style: might be nice to replace `){` with `) {` in this file for consistency with project standards. (Would also be nice to get eslint to do that for us sooner or later...)

Reply:
Done